src/ray/raylet/worker_pool.cc

```diff
@@ -173,8 +173,9 @@ void WorkerPool::SetNodeManagerPort(int node_manager_port) {
   node_manager_port_ = node_manager_port;
 }
 
-void WorkerPool::SetAgentManager(std::shared_ptr<AgentManager> agent_manager) {
-  agent_manager_ = agent_manager;
+void WorkerPool::SetRuntimeEnvAgentClient(
+    std::shared_ptr<RuntimeEnvAgentClient> runtime_env_agent_client) {
```

Comment:
Check nullptr here or just make it a pass by reference?

Reply:
node manager should outlive worker pool (it is like the entrypoint). Let's just use the reference instead?

Reply:
It's kind of hard. worker_pool_ initiates early in the member variables of a node_manager, and at its init time we don't yet have a runtime env agent client. If we use reference, at first we will have a dangling one which is confusing and hard to debug. So let's keep the pointer. It's also good for unit testing.